chain/client/src/stateless_validation/partial_witness/partial_witness_actor.rs

```diff
@@ -76,6 +78,7 @@ pub struct DistributeStateWitnessRequest {
     pub epoch_id: EpochId,
     pub chunk_header: ShardChunkHeader,
     pub state_witness: ChunkStateWitness,
+    pub contract_deploys: Vec<ContractCode>,
```

Comment:
When you speak about 

> attach the contract code-hashes to DistributeStateWitnessRequest

for code-hashes we mean code and hash, right? Not just the hash of the code

Reply:
I actually meant hash of the code, I will see if I can change to make it more clear.

Reply:
Sorry, why not use `Vec<CryptoHash>` to attach only the hash?

Reply:
In the first place we collect this data, we collect it as `ContractCode` (https://github.com/near/nearcore/blob/d73041cc1d1a70af4456fceefaceb1bf7f684fde/core/store/src/contract.rs#L32) with code itself, thus we are propagating the deployed code through to the point we send the message, otherwise we need to read the code again from the storage for each code hash.